pkg/aws/actuator/actuator.go

```diff
@@ -326,47 +336,96 @@ func (a *AWSActuator) sync(ctx context.Context, cr *minterv1.CredentialsRequest)
+			return fmt.Errorf("invalid codec, nil value")
 		}
-	case constants.PassthroughAnnotation:
-		logger.Debugf("provisioning with passthrough")
-		err := a.syncPassthrough(ctx, cr, credentialsRootSecret, logger)
+		logger.Info("actuator detected STS enabled cluster, enabling STS secret brokering for CredentialsRequests providing an IAM Role ARN")
+		awsSTSIAMRoleARN, err := awsSTSIAMRoleARN(a.Codec, cr)
 		if err != nil {
 			return err
 		}
-	case constants.MintAnnotation:
-		logger.Debugf("provisioning with cred minting")
-		err := a.syncMint(ctx, cr, logger)
+		if awsSTSIAMRoleARN != "" {
+			err = a.createSecret(awsSTSIAMRoleARN, cr.Spec.CloudTokenPath, cr.Spec.SecretRef.Name, cr.Spec.SecretRef.Namespace, logger)
+		}
 		if err != nil {
-			msg := "error syncing creds in mint-mode"
-			logger.WithError(err).Error(msg)
+			return err
+		}
+	} else {
+		credentialsRootSecret, err := a.GetCredentialsRootSecret(ctx, cr)
+		if err != nil {
+			logger.WithError(err).Error("issue with cloud credentials secret")
+			return err
+		}
+		switch credentialsRootSecret.Annotations[constants.AnnotationKey] {
+		case constants.InsufficientAnnotation:
+			msg := "cloud credentials insufficient to satisfy credentials request"
+			logger.Error(msg)
+			return &actuatoriface.ActuatorError{
+				ErrReason: minterv1.InsufficientCloudCredentials,
+				Message:   msg,
+			}
+		case constants.PassthroughAnnotation:
+			logger.Debugf("provisioning with passthrough")
+			err := a.syncPassthrough(ctx, cr, credentialsRootSecret, logger)
+			if err != nil {
+				return err
+			}
+		case constants.MintAnnotation:
+			logger.Debugf("provisioning with cred minting")
+			err := a.syncMint(ctx, cr, logger)
+			if err != nil {
+				msg := "error syncing creds in mint-mode"
+				logger.WithError(err).Error(msg)
+				return &actuatoriface.ActuatorError{
+					ErrReason: minterv1.CredentialsProvisionFailure,
+					Message:   fmt.Sprintf("%v: %v", msg, err),
+				}
+			}
+		default:
+			msg := fmt.Sprintf("unexpected value or missing %s annotation on admin credentials Secret", constants.AnnotationKey)
+			logger.Info(msg)
 			return &actuatoriface.ActuatorError{
 				ErrReason: minterv1.CredentialsProvisionFailure,
-				Message:   fmt.Sprintf("%v: %v", msg, err),
+				Message:   msg,
 			}
 		}
-	default:
-		msg := fmt.Sprintf("unexpected value or missing %s annotation on admin credentials Secret", constants.AnnotationKey)
-		logger.Info(msg)
-		return &actuatoriface.ActuatorError{
-			ErrReason: minterv1.CredentialsProvisionFailure,
-			Message:   msg,
-		}
 	}
+	return nil
+}
 
+// createSecret makes a time-based token available in a Secret in the namespace of an operator that
+// has supplied the following in the CredentialsRequest:
+// a non-nil spec.CloudTokenString
+// a path to the JWT token: spec.cloudTokenPath
+// a spec.SecretRef.Name
+// a cr.Spec.SecretRef.Namespace
+func (a *AWSActuator) createSecret(awsSTSIAMRoleARN string, cloudTokenPath string, secretRef string,
```

Comment:
nit: probably best to name this something STS-specific

Reply:
Yeah I agree, renaming in upcoming commit.